gensim/models/atmodel.py

```diff
@@ -460,11 +460,12 @@ def inference(self, chunk, author2doc, doc2author, rhot, collect_sstats=False, c
                 # make sure the term IDs are ints, otherwise np will get upset
                 ids = [int(idx) for idx, _ in doc]
             else:
-                ids = [idx for idx, _ in doc]
-            cts = np.array([cnt for _, cnt in doc])
+                ids = [id for id, _ in doc]
+            ids = np.array(ids, dtype=np.integer)
+            cts = np.array([cnt for _, cnt in doc], dtype=np.integer)
```

Comment:
I'm not familiar with this `np.integer` type. How does it differ from normal `np.int`? What's the difference?

Reply:
No difference in our case
```python
import numpy as np

arr1, arr2 = [1, 2, 3], []

assert np.array(arr1, dtype=np.int).dtype == \
       np.array(arr1, dtype=np.integer).dtype == \
       np.array(arr2, dtype=np.int).dtype == \
       np.array(arr2, dtype=np.integer).dtype
```

Reply:
all of it "casted" to `int64` on my x64 linux